Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/ProgressIndicators/ProgressIndicatorOrbsRotator.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Threading.Tasks;
+using TMPro;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.SceneTransitions
+{
+    /// <summary>
+    /// This class manages the 'rotating circle of dots' effect
+    /// that is used as a Progress Indicator effect.
+    /// </summary>
+    public class ProgressIndicatorOrbsRotator : MonoBehaviour, IProgressIndicator
+    {
+        public Transform MainTransform { get { return transform; } }
+        public ProgressIndicatorState State { get { return state; } }
+        public float Progress { set { progress = value; } }
+        public string Message { set { messageText.text = value; } }
+
+        [SerializeField]
+        private GameObject[] orbs = null;
+        
+        [SerializeField]
+        private TextMeshPro messageText = null;
+
+        [SerializeField]
+        private ProgressIndicatorState state = ProgressIndicatorState.Closed;
+        
+        public float RotationSpeedRawDegrees;
+        public float SpacingDegrees;
+        public float Acceleration;
+        public int Revolutions;
+        public bool TestStop = false;
+        public bool HasAnimationFinished = false;
+
+        private float progress;
+        private float timeElapsed;
+        private int deployedCount;
+        private bool timeUpdated;
+        private float[] angles;
+        private float timeSlice;
+        private float deg2rad = Mathf.PI / 180.0f;
```

Comment:
Seems like a useful thing that could exist in like, a MathUtils file... if not, maybe a private const? I don't think that deg2rad should change (it's just a conversion constant)